pkg/collector/corechecks/kubernetes/kubelet/provider/pod/provider.go

```diff
@@ -0,0 +1,272 @@
+			sender.Gauge(common.KubeletMetricsPrefix+"containers."+key+".terminated", 1, "", termTags)
+		}
+		if state.Waiting != nil && slices.Contains(whitelistedContainerStateReason["waiting"], strings.ToLower(state.Waiting.Reason)) {
+			waitTags := utils.ConcatenateStringTags(tags, "reason:"+strings.ToLower(state.Waiting.Reason))
+			sender.Gauge(common.KubeletMetricsPrefix+"containers."+key+".waiting", 1, "", waitTags)
+		}
+	}
+}
+
+type runningAggregator struct {
+	runningPodsCounter       map[string]float64
+	runningPodsTags          map[string][]string
+	runningContainersCounter map[string]float64
+	runningContainersTags    map[string][]string
+	podHasRunningContainers  map[string]bool
+}
+
+func newRunningAggregator() *runningAggregator {
+	return &runningAggregator{
+		runningPodsCounter:       make(map[string]float64),
+		runningPodsTags:          make(map[string][]string),
+		runningContainersCounter: make(map[string]float64),
+		runningContainersTags:    make(map[string][]string),
+		podHasRunningContainers:  make(map[string]bool),
+	}
+}
+
+func (r *runningAggregator) recordContainer(p *Provider, pod *kubelet.Pod, cStatus *kubelet.ContainerStatus, containerId string) {
+	if cStatus.State.Running == nil || time.Time.IsZero(cStatus.State.Running.StartedAt) {
+		return
+	}
+	r.podHasRunningContainers[pod.Metadata.UID] = true
+	tags, _ := tagger.Tag(containerId, collectors.LowCardinality)
+	if len(tags) == 0 {
+		return
+	}
+	hashTags := generateTagHash(tags)
+	r.runningContainersCounter[hashTags] += 1
+	if _, ok := r.runningContainersTags[hashTags]; !ok {
+		r.runningContainersTags[hashTags] = utils.ConcatenateTags(tags, p.config.Tags)
+	}
+}
+
+func (r *runningAggregator) recordPod(p *Provider, pod *kubelet.Pod) {
+	if !r.podHasRunningContainers[pod.Metadata.UID] {
+		return
+	}
+	podId := pod.Metadata.UID
+	if podId == "" {
+		log.Debug("skipping pod with no uid")
+		return
+	}
+	tags, _ := tagger.Tag(fmt.Sprintf("kubernetes_pod_uid://%s", podId), collectors.LowCardinality)
+	if len(tags) == 0 {
+		return
+	}
+	hashTags := generateTagHash(tags)
+	r.runningPodsCounter[hashTags] += 1
+	if _, ok := r.runningPodsTags[hashTags]; !ok {
+		r.runningPodsTags[hashTags] = utils.ConcatenateTags(tags, p.config.Tags)
+	}
+}
+
+func (r *runningAggregator) generateRunningAggregatorMetrics(sender aggregator.Sender) {
+	for hash, count := range r.runningContainersCounter {
+		sender.Gauge(common.KubeletMetricsPrefix+"containers.running", count, "", r.runningContainersTags[hash])
+	}
+	for hash, count := range r.runningPodsCounter {
+		sender.Gauge(common.KubeletMetricsPrefix+"pods.running", count, "", r.runningPodsTags[hash])
+	}
+}
+
+func parseQuantity(value string) (float64, error) {
+	var number, unit string
+	for _, char := range value {
+		if unicode.IsDigit(char) || char == '.' {
+			number += string(char)
+		} else {
+			unit += string(char)
+		}
+	}
```

Comment:
Can we use the K8s [parser](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/quantity.go) here? It seems K8s quantities can be bit more complicated than this and we already have a dependency on the `apimachinery` package.

Reply:
Today I learned that this exists, and it looks like it is already being used in other places in the code. I will spend some time playing around with it and trying to use it instead of this function